File: orange_hrm_ts/tests/steps/login.steps.ts
import { Page, expect } from '@playwright/test';
import { LoginPage } from '../pages/LoginPage';
import { DashboardPage } from '../pages/DashboardPage';

export async function acessarLogin(page: Page) {
  const login = new LoginPage(page);
  await login.goto();
}

export async function loginValido(page: Page) {
  const login = new LoginPage(page);
  await login.login('Admin', 'admin123');
}

export async function loginInvalido(page: Page) {
  const login = new LoginPage(page);
  await login.login('Admin', 'senha123');
}

export async function validarDashboard(page: Page) {
  const dash = new DashboardPage(page);
  await dash.isLoaded();
}

export async function validarErro(page: Page) {
  const login = new LoginPage(page);

  await page.waitForSelector('.oxd-alert-content-text', { timeout: 10000 });

  await expect(login.getError()).toBeVisible();
}

File: orange_hrm_python/tests/steps/login_steps.py
from playwright.sync_api import expect
from tests.pages.login_page import LoginPage

def acessar_login(page):
    LoginPage(page).goto()

def login_valido(page):
    LoginPage(page).login("Admin", "admin123")

def login_invalido(page):
    LoginPage(page).login("Admin", "errado")

def validar_erro(page):
    expect(LoginPage(page).get_error()).to_be_visible()